tests/fft.hpp

```diff
@@ -214,3 +216,87 @@ static void test_fft()
     ASSERT_LE(criterion, epsilon);
     ASSERT_LE(criterion2, epsilon);
 }
+
+template <typename ExecSpace, typename MemorySpace, typename Tin, typename Tout, typename X>
+static void test_fft_norm(ddc::FFT_Normalization const norm)
+{
+    DDom<DDim<X>> const x_mesh
+            = DDom<DDim<X>>(ddc::init_discrete_space(DDim<X>::
+                                                             init(ddc::Coordinate<X>(-1. / 4),
+                                                                  ddc::Coordinate<X>(1. / 4),
+                                                                  DVect<DDim<X>>(2))));
+    ddc::Chunk _f = ddc::Chunk(x_mesh, Allocator<MemorySpace, Tin>());
+    ddc::ChunkSpan f = _f.span_view();
+    ddc::for_each(
+            policy<ExecSpace>(),
+            ddc::get_domain<DDim<X>>(f),
+            DDC_LAMBDA(DElem<DDim<X>> const e) { f(e) = static_cast<Tin>(1); });
+
+    ddc::init_fourier_space<X>(x_mesh);
+    DDom<DFDim<ddc::Fourier<X>>> const k_mesh
+            = ddc::FourierMesh(x_mesh, is_complex<Tin>::value && is_complex<Tout>::value);
+
+    ddc::Chunk _f_bis = ddc::Chunk(ddc::get_domain<DDim<X>>(f), Allocator<MemorySpace, Tin>());
+    ddc::ChunkSpan f_bis = _f_bis.span_view();
+    ddc::deepcopy(f_bis, f);
+
+    ddc::Chunk _Ff = ddc::Chunk(k_mesh, Allocator<MemorySpace, Tout>());
+    ddc::ChunkSpan Ff = _Ff.span_view();
+    ddc::fft(ExecSpace(), Ff, f_bis, {norm});
+    Kokkos::fence();
+
+    ddc::Chunk _f_host = ddc::Chunk(ddc::get_domain<DDim<X>>(f), ddc::HostAllocator<Tin>());
+    ddc::ChunkSpan f_host = _f_host.span_view();
+    ddc::deepcopy(f_host, f);
+#if 0
+    std::cout << "\n input:\n";
+    ddc::for_each(
+            ddc::policies::serial_host,
+            ddc::get_domain<DDim<X>>(f_host),
+            [=](DElem<DDim<X>> const e) {
+                (std::cout << coordinate(ddc::select<DDim<X>>(e)))
+                        << "->" << f_host(e) << ", ";
+            });
+#endif
+
+    ddc::Chunk _Ff_host
+            = ddc::Chunk(ddc::get_domain<DFDim<ddc::Fourier<X>>>(Ff), ddc::HostAllocator<Tout>());
+    ddc::ChunkSpan Ff_host = _Ff_host.span_view();
+    ddc::deepcopy(Ff_host, Ff);
+#if 0
+    std::cout << "\n output:\n";
+    ddc::for_each(
+            ddc::policies::serial_host,
+            ddc::get_domain<DFDim<ddc::Fourier<X>>>(Ff_host),
+            [=](DElem<DFDim<ddc::Fourier<X>>> const e) {
+                (std::cout << coordinate(ddc::select<DFDim<ddc::Fourier<X>>>(e)))
+                        << "->" << Kokkos::abs(Ff_host(e)) << " "
+                        << ", ";
+            });
+#endif
+
+    double Ff0_expected;
+    switch (norm) {
+    case ddc::FFT_Normalization::OFF:
+        Ff0_expected = 2;
```

Comment:
Is it 2 because you have 2 points in your mesh ?

Reply:
Yes

Reply:
Ok then could you relate the two numbers by defining a variable ?

Reply:
That's the sum of elements of the input. Here f_1 + f_2 = 1+1 = 2.

`Ff0_expected = Ff(Ff.domain().front()) + Ff(Ff.domain.back())` ?

Reply:
I will commit that. If you change the `2` you also have to change the boundaries though:

`init(ddc::Coordinate<X>(-3. / 8), ddc::Coordinate<X>(3. / 8), DVect<DDim<X>>(4))));`

Reason is : for both configurations the FFT treats the domain [-1/2;1/2]. The boundaries of the mesh (-1/4 and 1/4, or -3/8 and 3/8) are the centers of first and last cells.